### PR TITLE
CORDA-641: Move special case handling of notary transactions into `SignedTransaction`

### PR DESCRIPTION
**Pull request by rnicoll**
Move special case handling of notary transactions into `SignedTransaction` so it's implemented once and reusable, instead of scattered through flows.

### CODE REVIEW

**Comment by josecoll** [comment]
Please can you make this PR clear that it relates to the issue report by https://r3-cev.atlassian.net/browse/CORDA-641:
**Contract upgrade transactions are verified incorrectly**

**Comment by adagys** [comment]
Maybe let's hold off with this until we get the contract upgrade transactions sorted

**Comment by rnicoll** [comment]
This is a worked example of some changes being discussed as part of #1670 and also for the upgrade transactions, to simplify both changes. However yes, we need solutions and then can look at how this fits in (potentially rebasing both off this).

**Comment by rnicoll** [comment]
Waiting on integration tests re-running (attachment loader failed, but known flaky), but this is ready for re-review.

File: core/src/main/kotlin/net/corda/core/transactions/SignedTransaction.kt [hunk 1]
```diff
@@ -180,6 +180,30 @@ data class SignedTransaction(val txBits: SerializedBytes<CoreTransaction>,
...
+    }
+
+    /**
+     * Resolves the underlying transaction with signatures and then returns it, handling any special case transactions
+     * such as [NotaryChangeWireTransaction].
+     */
+    fun resolveTransactionWithSignatures(services: ServiceHub): TransactionWithSignatures {
+        return when (transaction) {
+            is NotaryChangeWireTransaction -> resolveNotaryChangeTransaction(services)
+            is TraversableTransaction -> this
```
Comment (adagys) [inline core/src/main/kotlin/net/corda/core/transactions/SignedTransaction.kt]: Should probably be WireTransaction (doesn't really matter much in this case, but makes it more clear)

Comment (rnicoll) [reply core/src/main/kotlin/net/corda/core/transactions/SignedTransaction.kt]: Could do, but then means a separate `is FilteredTransaction ->`, is that an improvement overall?

Comment (adagys) [reply core/src/main/kotlin/net/corda/core/transactions/SignedTransaction.kt]: `FilteredTransaction` should never really be wrapped in a SignedTransaction, we don't want someone accidentally storing it. Hmm, maybe we should enforce that somwhere

**Pull request merged**